spring-integration-core/src/main/java/org/springframework/integration/transformer/SyslogToMapTransformer.java

```diff
@@ -88,6 +90,10 @@ else if (isString) {
 	private Map<String, ?> transform(String payload) {
 		Map<String, Object> map = new LinkedHashMap<String, Object>();
 		Matcher matcher = pattern.matcher(payload);
+		SimpleDateFormat dateFormat = dateFormats.poll();
+		if (dateFormat == null) {
+			dateFormat = new SimpleDateFormat("MMM dd HH:mm:ss");
```

Comment:
I think lazy initialization is fine.

I thought about having a limit on the queue so the offer would fail when full but then we end up with another configuration.

Plus these objects are used for a very short time; if the volume was that great, we'd be OOM well before now.